### ISSUE TITLE
ARBGetSubscriptionListResponse#subscriptionDetails returning string, not array

### ISSUE THREAD

**Issue by cjolly** [AuthorizeNet/sdk-ruby#70]
Hello, 
I'm trying to use `v1.8.6` to get a list of ARB subscriptions. I'm using the sample code from the [api docs](http://developer.authorize.net/api/reference/index.html#recurring-billing-get-a-list-of-subscriptions), however when I try and iterate on `subscriptionDetails`, it fails, as it is returning a `String`. The [schema](https://github.com/AuthorizeNet/sdk-ruby/blob/8021abea8d6f7073f47fdb9acdae887fde1d1004/lib/authorize_net/api/schema.rb#L3837) appears, at least in comments, to be correct. I'm not familiar at all with `ROXML`,so  that's where my troubleshooting ended up. 

```ruby
response = transaction.get_subscription_list(request)

if response.messages.resultCode == MessageTypeEnum::Ok
  p response.class
  p "Subscription Count: #{response.totalNumInResultSet}"
  p response.subscriptionDetails.class

  response.subscriptionDetails.each do |subscriptonDetail|
    p subscriptionDetail
  end
# snip
```

output: 
```txt
AuthorizeNet::API::ARBGetSubscriptionListResponse
"Subscription Count: 3"
String
NoMethodError: undefined method `each' for #<String:0x007f8f92061368>
```

EDIT: 

I also tried the master branch at 248cc52857c71f6d50f7253bcdca9c6ee34ca4a7, same result.

**Comment by namanbansal** [comment]
Hi @cjolly ,

I have sent a PR to fix the issue. I have also updated the sample code for get-list-of-subscription in the [sample-code-ruby repo](https://github.com/AuthorizeNet/sample-code-ruby/pull/24).

Naman

**Comment by cjolly** [comment]
Hello @namanbansal, 
Thanks! Your fix works great, I just tried out your branch.

**Comment by brianmc** [comment]
Merged into master, tagged release 1.8.7 (https://github.com/AuthorizeNet/sdk-ruby/releases/tag/1.8.7) and published new gem https://rubygems.org/gems/authorizenet/versions/1.8.7 

Thanks @namanbansal
